src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb

```diff
@@ -2594,11 +2594,17 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         End Function
 
         Protected Overrides Function CommonCreateTupleTypeSymbol(elementTypes As ImmutableArray(Of ITypeSymbol), elementNames As ImmutableArray(Of String)) As INamedTypeSymbol
-            Throw New NotSupportedException(VBResources.TuplesNotSupported)
+            Dim vbElementTypes = elementTypes.SelectAsArray(Function(t) t.EnsureVbSymbolOrNothing(Of TypeSymbol)(NameOf(elementTypes)))
+
+            Return TupleTypeSymbol.Create(locationOpt:=Nothing, elementTypes:=vbElementTypes, elementLocations:=Nothing, elementNames:=elementNames, compilation:=Me)
         End Function
 
         Protected Overrides Function CommonCreateTupleTypeSymbol(underlyingType As INamedTypeSymbol, elementNames As ImmutableArray(Of String)) As INamedTypeSymbol
-            Throw New NotSupportedException(VBResources.TuplesNotSupported)
+            Return TupleTypeSymbol.Create(
```

Comment:
C# also has checks here, such as `underlyingType` needs to be tuple-compatible. Maybe add PROTOTYPE markers to remember to add back in?